src/draft-blockchain.js

```diff
@@ -0,0 +1,136 @@
+const UTXO = require('UTXO')
+const Block = require('block')
+const mining_target = require('CONFIG')
+const Transaction = require('transaction')
+const genesis_coinbase = require('genesis')
+
+let current_blockchain
+
+const get_blockchain = () => {
+  if (current_blockchain === undefined) {
+    current_blockchain = new Blockchain()
+  }
+  return current_blockchain
+}
+
+class Blockchain{
+  constructor(){
+    this.chain = [genesis_coinbase]
+  }
+
+
+  insertBlock(block) {
+    if (!(block instanceof Block)) {
+      return false;
+    }   
+    for (let tx of block.transactions) {
+      if (!tx.isValid()) {
+        return false;
+      }
+      if (tx instanceof Transaction) {
+        for (let utxo of tx.utxos) {
+          if (!this.isValidUTXO(utxo)) {
+            return false;
+          }
+        }
+      }
+    }
+    if (!this.checkAgainstTarget(mining_target, block.getHash())) {
+      return false;
+    }
+    this.chain.push(block);
+    return true;
+  }
+
+
+  isChainValid(){
+    for(let i = 1; i < this.chain.length; i++){
+      const currentBlock = this.chain[i];
+      const previousBlock = this.chain[i - 1]
+
+      if(!currentBlock.hasValidTransactions()){
+        return false
+      }   
+      if(currentBlock.hash != currentBlock.calculateHash()){
+        return false;
+      }
+      if(currentBlock.previousHash != previousBlock.hash){        
+        return false;
+      }
+    return true
+    }
+  }
+ 
+
+  getLatestBlock(){
+    return this.chain[this.chain.length - 1]
+  }
+
+
+  checkAgainstTarget(mining_target, hash) {
+    return hash.substring(0, mining_target) !== Array(mining_target + 1).join("0") ? true : false;   
+  } 
```

Comment:
Looks like this method exists in other PR, we can probably revisit duplications later. Suggestion: 

```suggestion
  checkAgainstTarget(miningTarget, hash) {
    const targetPrefix = '0'.repeat(miningTarget);
    return hash.startsWith(targetPrefix);
  }
```